front/src/modules/pathfinding/hook/usePathfinding.ts

```diff
@@ -284,16 +301,33 @@ export const usePathfindingV2 = (
                 compact(updatedPathSteps)
               );
 
-              setPathProperties?.({
-                electrifications,
-                geometry,
-                suggestedOperationalPoints,
-                allWaypoints,
-                length: pathfindingResult.length,
-                trackSectionRanges: pathfindingResult.track_section_ranges,
-              });
-
-              pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
+              if (pathfindingResult.status === 'success') {
+                pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
+                if (setPathProperties)
+                  setPathProperties({
+                    electrifications,
+                    geometry,
+                    suggestedOperationalPoints,
+                    allWaypoints,
+                    length: pathResult.length,
+                    trackSectionRanges: pathResult.track_section_ranges,
+                  });
+              } else {
+                if (setPathProperties)
+                  setPathProperties({
+                    electrifications,
+                    geometry,
+                    suggestedOperationalPoints,
+                    allWaypoints,
+                    length: pathResult.length,
+                    incompatibleConstraints: pathfindingResult.incompatible_constraints,
+                    trackSectionRanges: pathResult.track_section_ranges,
+                  });
+                pathfindingDispatch({
+                  type: 'PATHFINDING_INCOMPATIBLE_CONSTRAINTS',
+                  message: `pathfindingErrors.${pathfindingResult.status}`,
+                });
+              }
```

Comment:
Maybe we could do like this instead of using two time the `setPathProperties` ?

```suggestion
              if (setPathProperties)
                setPathProperties({
                  electrifications,
                  geometry,
                  suggestedOperationalPoints,
                  allWaypoints,
                  length: pathResult.length,
                  incompatibleConstraints:
                    pathfindingResult.status === 'incompatible_constraints'
                      ? pathfindingResult.incompatible_constraints
                      : undefined,
                  trackSectionRanges: pathResult.track_section_ranges,
                });

              if (pathfindingResult.status === 'incompatible_constraints') {
                pathfindingDispatch({
                  type: 'PATHFINDING_INCOMPATIBLE_CONSTRAINTS',
                  message: `pathfindingErrors.${pathfindingResult.status}`,
                });
              } else {
                pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
              }
```